pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -167,24 +216,28 @@ func FindCandidates(ctx context.Context, cs kubernetes.Interface, state *framewo
 	if len(potentialNodes) == 0 {
 		klog.V(3).Infof("Preemption will not help schedule pod %v/%v on any node.", pod.Namespace, pod.Name)
 		// In this case, we should clean-up any existing nominated node name of the pod.
-		if err := util.ClearNominatedNodeName(cs, pod); err != nil {
+		if err := util.ClearNominatedNodeName(pl.fh.ClientSet(), pod); err != nil {
 			klog.Errorf("Cannot clear 'NominatedNodeName' field of pod %v/%v: %v", pod.Namespace, pod.Name, err)
 			// We do not return as this error is not critical.
 		}
 		return nil, nil
 	}
+
+	pdbs, err := getPodDisruptionBudgets(pl.pdbLister)
+	if err != nil {
+		return nil, err
+	}
+
+	offset, numCandidates := pl.updateAndReturnOffset(int32(len(potentialNodes)))
```

Comment:
would it be simpler to pick a random start point rather than making *DefaultPreemption stateful?

Reply:
we did consider that, but the round robin-like approach guarantees that it will evenly distributes preemption across all nodes, whereas a random approach could result in confusing results to the end-user (e.g. the same node getting picked for preemption twice in a row)

but I totally see why a random starting index might be nice to have too -- this is something that's certainly worth exploring in the future once we have more feedback from users

Reply:
picking the same node twice isn't an issue (not doing so is not part of the contract), it could happen with the existing logic as well.

Reply:
>it could happen with the existing logic as well

picking the same node twice *in a row* could happen, but if that happens, it also guarantees that all other nodes have been tried

Reply:
making this stateful and self-modifying makes it a lot harder to reason about (especially in terms of thread safety and data races)

Reply:
Done. Switched to a random offset. There is no difference in throughput between random and moving offset in integration benchmarks (as expected).